pennylane/ops/qubit.py

```diff
@@ -972,6 +972,44 @@ def _matrix(cls, *params):
             list(range(len(pauli_word))),
         )
 
+    _generator = None
+
+    @property
+    def generator(self):
+        if self._generator is None:
+            pauli_word = self.parameters[1]
+
+            # Simplest case is if the Pauli is the identity matrix
+            if pauli_word == "I" * len(pauli_word):
+                self._generator = [np.eye(2 ** len(pauli_word)), -1 / 2]
```

Comment:
@trbromley: we should overhaul how generators work. Rather than returning an operation class or a matrix, we should instead allow returning an operation _instance_ or a matrix.

That way we could support returning `qml.Identity(wires=self.wires)` here, which avoids creating a large identity matrix.

Reply:
@josh146, totally agree!

Reply:
I'll turn this into a GitHub issue :+1: